fth/save-input.fth

```diff
@@ -0,0 +1,67 @@
+\ SAVE-INPUT and RESTORE-INPUT
+
+anew task-save-input.fth
+
+private{
+
+: save-buffer ( -- column source-id 2 ) >in @ source-id 2 ;
+
+: restore-column ( column -- flag )
+  source nip over < if drop true exit then
+  >in ! false
+;
+
+\ Return the file-position of the beginning of the current line in
+\ file SOURCE-ID.  Assume that the current line is stored in SOURCE
+\ and that the current file-position is at an end-of-line (or
+\ end-of-file).
+: line-start-position ( -- ud )
+  source-id file-position throw
+  \ unless at end-of-file, subtract newline
+  source-id file-size throw 2over d= 0= if 1 s>d d- then
+  \ subtract line length
+  source nip s>d d-
+;
+
+: save-file ( column line filepos:ud source-id 5 -- )
+  >in @
+  source-line-number@
+  line-start-position
+  source-id
+  5
+;
+
+: restore-file ( column line filepos:ud -- flag )
+  source-id reposition-file if 2drop true exit then
+  source-line-number!
+  refill 0= if drop true exit then
```

Comment:
In commit https://github.com/philburk/pforth/pull/25/commits/09d08b416a8385ad4e8395044d717809e18bafa7, I re-indented with 4 spaces. 

For my own projects I like to imitate the style that Mitch Bradley uses in [Open Firmware](https://github.com/MitchBradley/openfirmware), except that I use 2 spaces for indentation instead of 3. Mitch Bradley's code is IMO very clear.  The style in pForth is of course different and I'm happy to use whatever you prefer.

I'm now just writing down some of my thoughts on style (feel free to disagree :-):
* I'm no fan of uppercase keywords, as it's harder to type and my editor already highlights the keywords. I write almost everything in lowercase. An exception are constants like `SEEK_CUR` that are the same as in C. Also in comments, an uppercase name refers to a variable (to avoid quoting).
* I use `-` to combine words like `foo-bar` instead of `foo.bar` or `foo_bar`.  As all the standard words use `-`  this looks more consistent. 
* As you've noticed, I use EXIT frequently. IMO that's more readable than nested IFs.  I even prefer EXIT to terminate loops over WHILE.  Loop constructs like `BEGIN  ... WHILE  ... WHILE ... REPEAT THEN` are confusing. (And difficult to recongnize for editors.)
* Usually I avoid non-letter characters like `>$&#%` in names and use words instead. Exceptions
are `foo?` which is a predicate, `foo>bar` which means "move or convert `foo` to `bar`" and `/foo` is the size (usually in bytes) of struct `foo`.  In the past I used `(foo)` for internal/private helpers, but these days I prefer `%foo` as it is shorter and `%%foo` looks better than `((foo))`.
* I'm still undecided whether `;` should be on same line as the last word in a colon definition or on it's own line.